### DISCUSSION TITLE
[FR] Humidifer/Dehumidifer/Generic hygrostat

### DISCUSSION THREAD

**Discussion by jondavidsson** [Clooos/Bubble-Card#936] [Feature requests]
It would be great with a card for Humidifer/Dehumidifer/Generic hygrostat similar to the climate :)

Thanks for your awesome work!

**Comment by Clooos** [comment]
Hi! I have no idea of what this could looks like and what are the services that are required. Could you provide more informations on that?

**Comment by jondavidsson** [comment]
I've only used the Generic hygrostat. This simply takes a humidity sensor and a switch and makes a virtual humidifier/dehumidifier. The card then have a on/off and a humidity target value. So if it's a dehumidifier and the target humidity is set to 50%, it turns on the switch if the sensor value is over 50%.

Available actions: humidifier.set_mode, humidifier.set_humidity, humidifier.turn_on, humidifier.turn_off, humidifier.toggle
https://www.home-assistant.io/integrations/humidifier/

**Reply by Clooos** [reply]
Can't you do it with a switch button? You can then modify the button action and also add a sub-button to show the humidity for example.

**Reply by jondavidsson** [reply]
Actually I haven't tried. At the moment I use Mushroom for the humidifier. My initial thought was that the cards seems to have alot in common, so maybe most of the work is already done. At the moment I use Mushroom for this card, but I might have a go with your suggestion about the switch. Don't know if it's possible to have a sub button with " - 50% + "? Where I can set the percentage by clicking + or minus like in the climate card?

**Reply by madkatz01** [reply]
Unfortunately using the regular button doesn't work because the humidifier has attributes similar to climate (like mode: auto, manual, off) and target humidity (0 to 100%). The sub-button can display an attribute list, but not let you select the mode. Nor can we adjust the target.

**Comment by tapp1993** [comment]
Jumping in to support this. It would be great if it could work somewhat like the climate card. With the ability to toggle, view the current humidity and set the target humidity.
